### PR TITLE
Use std::move

### PR DESCRIPTION
**Pull request by ShFil119**
One more emplace_back and usage std::move when possible.

- [x] remove unneeded std::move (POD/ptr)
- [x] avoid temporary zd
- [x] cleanup

### CODE REVIEW

File: rwengine/src/engine/SaveGame.cpp [hunk 1]
```diff
@@ -985,7 +985,7 @@ bool SaveGame::loadGame(GameState& state, const std::string& file) {
         // @toodo restore gang density
         zd.pedGroupDay = day.pedgroup;
         zd.pedGroupNight = night.pedgroup;
-        gamezones.push_back(zd);
+        gamezones.push_back(std::move(zd));
```
Comment (danhedron) [inline rwengine/src/engine/SaveGame.cpp]: This could be done with emplace_back, avoiding the temporary ``zd``.

**Comment by danhedron** [comment]
A few of these are just PODs, where moving is the same as copying. You can just use ``move`` on the types that contain strings, vectors and other allocating types.

**Comment by CookiePLMonster** [comment]
Does it matter if they are PODs? It's still a good practice, and emplacing is beneficial for pretty much any type (except fundamental types).

File: rwlib/source/data/Clump.hpp [hunk 2]
```diff
@@ -149,6 +149,11 @@ struct Geometry {
         std::string name;
         std::string alphaName;
         TextureData::Handle texture;
+        Texture(std::string _name, std::string _alphaName, TextureData::Handle _texture)
+            : name(_name)
+            , alphaName(_alphaName)
```
Comment (CookiePLMonster) [inline rwlib/source/data/Clump.hpp]: I believe those don't implicitly move - you might need to `std::move` them inside the initializer list.

File: rwlib/source/loaders/LoaderDFF.cpp [hunk 3]
```diff
@@ -359,7 +359,7 @@ void LoaderDFF::readTexture(Geometry::Material &material,
 
     TextureData::Handle textureinst =
         texturelookup ? texturelookup(name, alpha) : nullptr;
-    material.textures.push_back({name, alpha, textureinst});
+    material.textures.emplace_back(name, alpha, textureinst);
```
Comment (CookiePLMonster) [inline rwlib/source/loaders/LoaderDFF.cpp]: You should be using `std::move` here too, as now you're passing your strings by value. Looked up the full function and looks like moving those strings away is perfectly safe.

File: rwlib/source/data/Clump.hpp (hunk 2 above)
Comment (ShFil119) [reply rwlib/source/data/Clump.hpp]: I will try to use perfect forwarding later. Having quite busy week. ;)

Comment (CookiePLMonster) [reply rwlib/source/data/Clump.hpp]: Cool =) Though in this case you don't need perfect forwarding - `std::move` alone does fine and is the best approach no matter if you're moving or copying strings into this constructor.

**Pull request merged**